misc/init.d/cfengine3.in

```diff
@@ -348,10 +348,18 @@ cf_stop_daemon()
         fi
 
         pidfile_pid="$(cf_pidfile_pid "$pidfile")"
```

Comment:
before this commit, this construction:
```
         pidfile_pid="$(cf_pidfile_pid "$pidfile")"
         if [ -n "$pidfile_pid" ]; then
             pids="$pidfile_pid"
         else
             pids="$(cf_get_matching_pids "$base_daemon")"
         fi
```
repeated itself in this function three times. But I see you've edited it only in one place. Why not edit it in other places or rewrite the function so it was only once?

I might be missing something, but have you considered rewriting this loop so instead of this:
```
get_pids_from_pidfile_or_get_matching_pids
while ... do
   if iter > 1 then
      sleep 2
      get_pids_from_pidfile_or_get_matching_pids
   fi
   send_signal
   get_pids_from_pidfile_or_get_matching_pids
done
```

it was more like this?
```
while ... do
   get_pids_from_pidfile_or_get_matching_pids
   send_signal
   sleep 2
done
```

Reply:
We don't want to `sleep 2` if that's not necessary. With the iteration counter there's a chance the process could terminate quickly with no `sleep` required.